src/client_side_reply.cc

```diff
@@ -458,9 +458,18 @@ clientReplyContext::handleIMSReply(StoreIOBuffer result)
         http->logType.update(LOG_TCP_REFRESH_UNMODIFIED);
         http->request->flags.staleIfHit = false; // old_entry is no longer stale
```

Comment:
Do we have to do these two operations before calling updateOnNotModified()? With the old unconditional code, the order did not look suspicious. Now, since we are essentially doing `if (cannot update) process as a miss`, these two adjustments feel premature. If updateOnNotModified() code path does not use logType and flags.staleIfHit, then I would reorder.